../src/desktop.c:17:10: fatal error: 'wlr/types/wlr_gtk_primary_selection.h' file not found
#include <wlr/types/wlr_gtk_primary_selection.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../tests/test-xdg-shell.c:7:
In file included from ../tests/testlib.h:6:
In file included from ../src/server.h:14:
../src/desktop.h:11:10: fatal error: 'wlr/types/wlr_gtk_primary_selection.h' file not found
#include <wlr/types/wlr_gtk_primary_selection.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

--- src/desktop.c.orig	2021-07-10 00:06:41 UTC
+++ src/desktop.c
@@ -14,7 +14,7 @@
 #include <wlr/types/wlr_data_control_v1.h>
 #include <wlr/types/wlr_export_dmabuf_v1.h>
 #include <wlr/types/wlr_gamma_control_v1.h>
-#include <wlr/types/wlr_gtk_primary_selection.h>
+#include <wlr/types/wlr_primary_selection_v1.h>
 #include <wlr/types/wlr_idle.h>
 #include <wlr/types/wlr_input_inhibitor.h>
 #include <wlr/types/wlr_layer_shell_v1.h>
@@ -600,7 +600,7 @@ phoc_desktop_constructed (GObject *object)
 						 WLR_SERVER_DECORATION_MANAGER_MODE_CLIENT);
   self->idle = wlr_idle_create(server->wl_display);
   self->primary_selection_device_manager =
-    wlr_gtk_primary_selection_device_manager_create(server->wl_display);
+    wlr_primary_selection_v1_device_manager_create(server->wl_display);
   self->input_inhibit =
     wlr_input_inhibit_manager_create(server->wl_display);
   self->input_inhibit_activate.notify = input_inhibit_activate;
--- src/desktop.h.orig	2021-07-10 00:06:41 UTC
+++ src/desktop.h
@@ -8,7 +8,7 @@
 #include <wlr/types/wlr_compositor.h>
 #include <wlr/types/wlr_foreign_toplevel_management_v1.h>
 #include <wlr/types/wlr_gamma_control_v1.h>
-#include <wlr/types/wlr_gtk_primary_selection.h>
+#include <wlr/types/wlr_primary_selection.h>
 #include <wlr/types/wlr_idle.h>
 #include <wlr/types/wlr_input_inhibitor.h>
 #include <wlr/types/wlr_input_method_v2.h>
@@ -68,7 +68,7 @@ struct _PhocDesktop {
 	struct wlr_export_dmabuf_manager_v1 *export_dmabuf_manager_v1;
 	struct wlr_server_decoration_manager *server_decoration_manager;
 	struct wlr_xdg_decoration_manager_v1 *xdg_decoration_manager;
-	struct wlr_gtk_primary_selection_device_manager *primary_selection_device_manager;
+	struct wlr_primary_selection_v1_device_manager *primary_selection_device_manager;
 	struct wlr_idle *idle;
 	struct wlr_input_inhibit_manager *input_inhibit;
 	struct wlr_layer_shell_v1 *layer_shell;
